Maple2.Model/Metadata/Table/ColorPaletteTable.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using Maple2.Model.Common;
+
+namespace Maple2.Model.Metadata;
+
+public record ColorPaletteTable(IReadOnlyDictionary<int, Dictionary<int, ColorPaletteTable.Entry>> Entries) : Table {
```

Comment:
Other `Dictionary` should also be `IReadOnlyDictionary`

Reply:
not sure if i would be able to do that with how i'm ingesting it.

Reply:
you should change the way you're ingesting it to make this possible (build a separate dictionary, then insert it once it's done being built). It's important that all metadata is fully immutable because a single instance is shared across the whole server.